src/main/java/Duke.java

```diff
@@ -1,10 +1,185 @@
+                System.out.println(target);
+                return InputHandleStatus.OK;
+            default:
+                Task newTask = parseTask(splitted);
+                taskList.add(newTask);
+                System.out.println("added: " + newTask);
+                return InputHandleStatus.OK;
+        }
+    }
+
+    private static void initialise() {
+        String greeting = "Hello! I'm Duke";
+        String assist = "What can I do for you?";
+        System.out.println(greeting);
+        System.out.println(assist);
+    }
+
+    private enum InputHandleStatus {
+        OK, END, ERROR
+    }
+
+    static class Task {
+
+        private String title = "";
+        private boolean doneStatus = false;
+
+        protected String getType() {
+            return "task";
+        }
+
+        private Task() {
+            // prevent uninitialised task
+        }
+
+        public Task(String title) {
+            this.doneStatus = false;
+            this.title = title;
+        }
+
+        public String getTitle() {
+            return title;
+        }
+
+        public void setTitle(String title) {
+            this.title = title;
+        }
+
+        public void setDoneStatus(boolean status) {
+            this.doneStatus = status;
+        }
+
+        public String getStatusIcon() {
+            return this.doneStatus ? "x" : " ";
+        }
+
+        public String toString() {
+            return "[" + Character.toUpperCase(this.getType().charAt(0)) + ']' + '[' + this.getStatusIcon() + ']' + ' ' + this.title;
+        }
+    }
+
+    public static class Todo extends Task {
+        public Todo(String description) {
+            super(description);
+        }
+    }
+
+    public static class Deadline extends Task {
+        String by;
+
+        public Deadline(String description, String by) {
+            super(description);
+            setBy(by);
+        }
+
+        public String getBy() {
+            return by;
+        }
+
+        public void setBy(String by) {
+            this.by = by;
+        }
+
+        @Override
+        public String toString() {
+            return super.toString() + " (by: " + getBy() + ")";
+        }
+
+        @Override
+        protected String getType() {
+            return "deadline";
+        }
+    }
+
+    public static class Event extends Task {
+        String to;
+
+        public Event(String description, String to) {
+            super(description);
+            setTo(to);
+        }
+
+        public String getTo() {
+            return to;
+        }
+
+        public void setTo(String to) {
+            this.to = to;
+        }
+
+        @Override
+        public String toString() {
+            return super.toString() + " (at: " + getTo() + ")";
+        }
+
+        @Override
+        protected String getType() {
+            return "event";
+        }
+    }
+
+    private static void finalise() {
```

Comment:
Could 'terminate' be a better choice than 'finalise'?

Reply:
I also weighed it and came across [this SO question](https://softwareengineering.stackexchange.com/questions/163004/what-is-the-opposite-of-initialize-or-init). Did not choose `terminate` since it sounds like it is the last thing while it might no longer be in future (e.g. might have some other cleanup after saying goodbye).